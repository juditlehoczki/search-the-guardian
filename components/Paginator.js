import React from "react";
import { View, Text, TouchableOpacity } from "react-native";

import s from "../styles.js";

const Paginator = (props) => {
  const { currentPage, pagesCount, changePage } = props;
  return (
    <View style={s.paginator}>
      <TouchableOpacity onPress={() => currentPage !== 1 && changePage(-1)}>
        <Text style={s.pageChangerButton}>&lt;</Text>
      </TouchableOpacity>
      <Text style={s.pages}>{`${currentPage}/${pagesCount}`}</Text>
      <TouchableOpacity
        disabled={currentPage === pagesCount}
        onPress={() => currentPage !== pagesCount && changePage(1)}
      >
        <Text style={s.pageChangerButton}>&gt;</Text>
      </TouchableOpacity>
    </View>
  );
};

export default Paginator;
