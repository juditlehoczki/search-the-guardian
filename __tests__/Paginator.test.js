import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {TouchableWithoutFeedback, Text} from 'react-native';

import Paginator from '../components/Paginator.js';

let fakeChangePage;
let props;

describe('Paginator Component', () => {
  beforeEach(() => {
    fakeChangePage = jest.fn();
    props = {
      pagesCount: 2,
      currentPage: 1,
      changePage: fakeChangePage,
    };
  });

  test('Snapshot - it renders correctly', () => {
    const snapshot = renderer.create(<Paginator {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  test('Current page and number of total pages are rendered correctly', () => {
    const fakePaginator = renderer.create(<Paginator {...props} />);
    const text = fakePaginator.root.findAllByType(Text);
    expect(text[1].props.children).toBe(
      `${props.currentPage}/${props.pagesCount}`,
    );
  });

  test('Previous button: calls changePage function when pressed and have previous pages available', () => {
    const fakePaginator = renderer.create(
      <Paginator {...props} currentPage={2} />,
    );
    const buttons = fakePaginator.root.findAllByType(TouchableWithoutFeedback);
    buttons[0].props.onPress();
    expect(fakeChangePage).toHaveBeenCalledWith(-1);
  });
  test('Previous button: does not call changePage function when pressed when there are no previous pages', () => {
    const fakePaginator = renderer.create(<Paginator {...props} />);
    const buttons = fakePaginator.root.findAllByType(TouchableWithoutFeedback);
    buttons[0].props.onPress();
    expect(fakeChangePage).not.toHaveBeenCalled();
  });

  test('Next button: calls changePage function when pressed and have further pages available', () => {
    const fakePaginator = renderer.create(<Paginator {...props} />);
    const buttons = fakePaginator.root.findAllByType(TouchableWithoutFeedback);
    buttons[1].props.onPress();
    expect(fakeChangePage).toHaveBeenCalledWith(1);
  });
  test('Next button: does not call changePage function when pressed when there are no further pages', () => {
    const fakePaginator = renderer.create(
      <Paginator {...props} currentPage={2} />,
    );
    const buttons = fakePaginator.root.findAllByType(TouchableWithoutFeedback);
    buttons[1].props.onPress();
    expect(fakeChangePage).not.toHaveBeenCalled();
  });
});
