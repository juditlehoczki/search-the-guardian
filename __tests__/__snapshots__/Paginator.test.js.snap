// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component Snapshot - it renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "fontSize": 18,
      "justifyContent": "center",
      "textAlign": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#D1529E",
          "borderRadius": 5,
          "borderStyle": "dotted",
          "borderWidth": 1,
          "color": "#D1529E",
          "fontSize": 20,
          "marginHorizontal": 5,
          "paddingHorizontal": 5,
          "textAlignVertical": "center",
        }
      }
    >
      &lt;
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#0F406E",
        "fontWeight": "bold",
      }
    }
  >
    1/2
  </Text>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderColor": "#D1529E",
          "borderRadius": 5,
          "borderStyle": "dotted",
          "borderWidth": 1,
          "color": "#D1529E",
          "fontSize": 20,
          "marginHorizontal": 5,
          "paddingHorizontal": 5,
          "textAlignVertical": "center",
        }
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;
